services/fxaccounts/FxAccountsKeys.jsm

```diff
@@ -71,7 +79,31 @@ class FxAccountsKeys {
    *        or null if no user is signed in
    */
   async getKeys() {
-    return this._fxia.withCurrentAccountState(async currentState => {
+    if (this._fxia.rustFxa) {
+      let {
+        key: { kid, k: kSync },
+      } = await this._fxia.rustFxa.getAccessToken(SCOPE_OLD_SYNC);
+      let kXCS = kid.split("-")[1];
+
+      // Base64 -> hex.
+      kSync = CommonUtils.bufferToHex(
+        new Uint8Array(
+          ChromeUtils.base64URLDecode(kSync, { padding: "ignore" })
+        )
+      );
+      kXCS = CommonUtils.bufferToHex(
+        new Uint8Array(ChromeUtils.base64URLDecode(kXCS, { padding: "ignore" }))
+      );
+
+      // In practice the users of getKeys() only use the `k...` keys.
+      return {
+        kSync,
+        kXCS,
+        kExtSync: null, // huuuuh
+        kExtKbHash: null, // yeah huuh.
```

Comment:
It's OK, these keys will go away with the landing of the new webext-storage work, which causes us to stop using special keys and start using the same `kSync` as all the other synced datatypes. So the fact that we don't have them will not be a problem.